### PR TITLE
REF: Pre-empt TypeError in groupby

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd @jreback, should be orthogonal to #29228.

### CODE REVIEW

**Comment by gfyoung** [comment]
Was this `TypeError` ever tested?  Is it possible to cover in the tests?

**Comment by jbrockmendel** [comment]
> Was this TypeError ever tested? Is it possible to cover in the tests?

The `pass` on L225 is currently reached in the tests.  The `raise` on L227 is not.

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -193,15 +193,20 @@ def apply(self, f, data, axis=0):
         group_keys = self._get_group_keys()
         result_values = None
 
-        # oh boy
-        f_name = com.get_callable_name(f)
-        if (
-            f_name not in base.plotting_methods
+        sdata = splitter._get_sorted_data()
+        if sdata.ndim == 2 and np.any(sdata.dtypes.apply(is_extension_array_dtype)):
```
Comment (jreback) [inline pandas/core/groupby/ops.py]: I would move this to the Splittler class something like

```
@cache_only
def has_extension_array(self):
     return self.data.dtypes.apply(...)
```

and can handle this on Series & Frame splitter

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: yah, this function is definitely getting refactored in a later pass.  id like to get the exceptions sorted out before trying to decide about the refactor

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
k looks fine (hopefully you have followons!). rebase and ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**